src/callback/callback_utils.cairo

```diff
@@ -18,25 +18,44 @@
 //                                  IMPORTS
 // *************************************************************************
 // Core lib imports.
+use core::keccak;
 use starknet::ContractAddress;
-use result::ResultTrait;
+use starknet::syscalls::emit_event_syscall;
 
 // Local imports.
 use satoru::data::data_store::{IDataStoreSafeDispatcher, IDataStoreSafeDispatcherTrait};
+use satoru::data::keys;
 use satoru::event::event_utils::EventLogData;
 use satoru::order::order::Order;
 use satoru::deposit::deposit::Deposit;
 use satoru::withdrawal::withdrawal::Withdrawal;
+use satoru::callback::order_callback_receiver::interface::{
+    IOrderCallbackReceiverSafeDispatcher, IOrderCallbackReceiverSafeDispatcherTrait
+};
+use satoru::callback::deposit_callback_receiver::interface::{
+    IDepositCallbackReceiverSafeDispatcher, IDepositCallbackReceiverSafeDispatcherTrait
+};
+use satoru::callback::withdrawal_callback_receiver::interface::{
+    IWithdrawalCallbackReceiverSafeDispatcher, IWithdrawalCallbackReceiverSafeDispatcherTrait
+};
 
-/// Validate that the callbackGasLimit is less than the max specified value.
+/// Validate that the callback_gas_limit is less than the max specified value.
 /// This is to prevent callback gas limits which are larger than the max gas limits per block
 /// as this would allow for callback contracts that can consume all gas.
 /// # Arguments
 /// * `data_store` - The data store to use.
 /// * `callback_gas_limit` - The callback gas limit.
-fn validate_callback_gas_limit(
-    data_store: IDataStoreSafeDispatcher, callback_gas_limit: u128
-) { // TODO
+fn validate_callback_gas_limit(data_store: IDataStoreSafeDispatcher, callback_gas_limit: u128) {
```

Comment:
not sure this method is actually needed as I'm not sure it's possible to specify gas limit when calling another contract?

Reply:
cant be done for the moment, see: https://github.com/starkware-libs/cairo/issues/4028#issuecomment-1710156964